contracts/DomainRegistryV2.sol

```diff
@@ -101,27 +170,28 @@ contract DomainRegistryV2 is Initializable, OwnableUpgradeable {
         if (parentDomainOwner == address(0)) {
             parentReward = 0;
         } else {
-            parentReward = price * REWARD_PERCENT_OWNER / 100;
+            parentReward = (price * REWARD_PERCENT_OWNER) / 100;
         }
         uint256 ownerReward = price - parentReward;
 
         domains[rootDomain] = msg.sender;
         shares[msg.sender] += ownerReward;
-        payable(parentDomainOwner).transfer(parentReward);
+        emit DomainRegistered(msg.sender, rootDomain);
 
+        payable(parentDomainOwner).transfer(parentReward);
```

Comment:
It is not the best approach to transfer ether in this place. What if parentDomainOwner is a contract, which revert the transaction when it get some ether? You contract would be blocked from registration in some cases. Much better to use the withdraw pattern here.
   -2 points

Reply:
fixed withdraw method, each domain controller can call it and withdraw its own money